### PR TITLE
Implemented All Basics SDK functionality, language functionalities + test cases 

### PR DESCRIPTION
**Pull request by piyush-gupta1999**
Terms:

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
> 
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: src/Morphir/IR/SDK/List.elm [hunk 1]
```diff
@@ -117,3 +122,90 @@ listType attributes itemType =
 construct : a -> Value ta a
 construct a =
     Value.Reference a (toFQName moduleName "construct")
+
+
+myMap : Native.Function
+myMap eval args =
```
Comment (AttilaMihaly) [inline src/Morphir/IR/SDK/List.elm:128]: Can we rename this to `nativeMap`?

File: src/Morphir/IR/SDK/List.elm [hunk 2]
```diff
@@ -117,3 +122,90 @@ listType attributes itemType =
...
+                            _ ->
+                                Err (UnexpectedArguments args)
+                    )
+
+        _ ->
+            Err (UnexpectedArguments args)
+
+
+append : Native.Function
+append eval args =
```
Comment (AttilaMihaly) [inline src/Morphir/IR/SDK/List.elm:153]: Can we rename this to `nativeAppend`?

File: src/Morphir/IR/SDK/Basics.elm [hunk 3]
```diff
@@ -180,6 +180,90 @@ nativeFunctions =
...
+                    _ ->
+                        Err (UnexpectedArguments [ arg1, arg2 ])
+            )
+      )
+    , ( "add"
+      , Native.binaryStrict
+            (\arg1 arg2 ->
+                case ( arg1, arg2 ) of
+                    ( Value.Literal _ (FloatLiteral v1), Value.Literal _ (FloatLiteral v2) ) ->
+                        Ok (Value.Literal () (FloatLiteral (v1 + v2)))
```
Comment (AttilaMihaly) [inline src/Morphir/IR/SDK/Basics.elm:192]: Can you add cases to cover mixed combinations like `Float + Int` and `Int + Float`? You will need to use `toFloat` on `Int` before you can add them and they should both return `Float`.

This applies to `add`, `subtract`, `multiply` and `lessThan`.

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: src/Morphir/IR/SDK/List.elm (hunk 1 above)
Comment (piyush-gupta1999) [reply src/Morphir/IR/SDK/List.elm:128]: Renamed

File: src/Morphir/IR/SDK/List.elm (hunk 2 above)
Comment (piyush-gupta1999) [reply src/Morphir/IR/SDK/List.elm:153]: Renamed

File: src/Morphir/IR/SDK/Basics.elm (hunk 3 above)
Comment (piyush-gupta1999) [reply src/Morphir/IR/SDK/Basics.elm:192]: Test cases added.

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**